### PR TITLE
DEXW-2726: Sign SetAssetScript

### PR DESCRIPTION
**Pull request by mrpq**
#### Не вливать, пока не вольется https://github.com/waves-exchange/provider-web/pull/3 и я не отребейзю эту ветку

### CODE REVIEW

File: src/iframe-entry/pages/SignIssue/SignIssueComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+    TabPanels,
+    Box,
+    Checkbox,
+    Label,
+    Copy,
+    ExternalLink,
+    iconIssueTransaction,
+} from '@waves.exchange/react-uikit';
+import { Help } from '../../components/Help/Help';
+import { AssetLogo } from '@waves.exchange/react-uikit/dist/esm/components/AssetLogo/AssetLogo';
```
Comment (finico) [inline src/iframe-entry/pages/SignIssue/SignIssueComponent.tsx]: duplicated import but from dist

File: src/iframe-entry/pages/SignIssue/SignIssueContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    );
+
+    return (
+        <SignIssueComponent
+            assetId={tx.id}
+            assetName={tx.name}
+            assetDescription={tx.description}
+            assetType={tx.reissuable ? 'Reissuable' : 'Non-reissuable'}
+            decimals={tx.decimals}
+            assetScript={tx.script || 'asdfasdfasdfasdfasdf '}
```
Comment (finico) [inline src/iframe-entry/pages/SignIssue/SignIssueContainer.tsx]: what a wonderful script :)

File: src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+import React, {
+    FC,
+    MouseEventHandler,
+    useState,
+    useCallback,
+    useEffect,
+} from 'react';
+import { Confirmation } from '../../components/Confirmation';
+import { TransactionDetails } from '../../components/TransactionDetails/TransactionDetails';
+import { AssetLogo } from '@waves.exchange/react-uikit/dist/esm/components/AssetLogo/AssetLogo';
```
Comment (finico) [inline src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx]: same for import

**Comment by mrpq** [comment]
@finico 
**fixed**

File: src/iframe-entry/components/Confirmation.tsx [hunk 4]
```diff
@@ -13,19 +13,24 @@ type IConfirmationProps = TFlexProps & {
...
+    canConfirm,
     name,
     balance,
     children,
     onReject,
     onConfirm,
     ...rest
 }) => {
+    const isConfirmDisabled =
+        typeof canConfirm === 'undefined' ? false : !canConfirm;
```
Comment (tsigel) [inline src/iframe-entry/components/Confirmation.tsx]: Why canConfirm may be undefined?

File: src/iframe-entry/components/Help/Help.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            arrowSize="4px"
+            hasArrow={true}
+            arrowColor="#5A81EA" // TODO научить тултип понимать цвета из темы
+            offset={4}
+            maxWidth="360px"
+            label={(): ReactNode => (
+                <Box
+                    borderTop="4px solid"
+                    borderTopColor="primary.$300"
+                    backgroundColor="#4a5060" // TODO уточнить этот цвет
```
Comment (tsigel) [inline src/iframe-entry/components/Help/Help.tsx]: Is this color from design?

File: src/iframe-entry/components/Help/Help.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+export const Help: FC<TooltipProps> = ({
+    children,
+    popperOptions,
+    ...rest
+}) => {
+    return (
+        <Tooltip
+            arrowSize="4px"
+            hasArrow={true}
+            arrowColor="#5A81EA" // TODO научить тултип понимать цвета из темы
```
Comment (tsigel) [inline src/iframe-entry/components/Help/Help.tsx]: Create task for this comment.

File: src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+                        borderRadius="circle"
+                        bg="rgba(129, 201, 38, 0.1)"
+                        height={60}
+                        width={60}
+                        flexShrink={0}
+                    >
+                        <Icon
+                            icon={iconSetAssetScript}
+                            size={40}
+                            color="#81C926"
```
Comment (tsigel) [inline src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx]: Theme?

File: src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+                                            example, if a regular tx is made for
+                                            a Smart Asset, the cost is 0.001 +
+                                            0.004 = 0.005 WAVES. If an exchange
+                                            transaction is made, the cost is
+                                            0.003 + 0.004 = 0.007 WAVES.
+                                            <ExternalLink
+                                                mt="$5"
+                                                variant="body2"
+                                                display="block"
+                                                href="https://docs.wavesplatform.com/en/building-apps/smart-contracts/smart-assets"
```
Comment (tsigel) [inline src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx]: External link has no target blank attribute by default?

Comment (mrpq) [reply src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx]: no it doesn't

File: src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx (hunk 7 above)
Comment (mrpq) [reply src/iframe-entry/pages/SignSetAssetScript/SignSetAssetScriptComponent.tsx]: Color is not in the theme

File: src/iframe-entry/components/Help/Help.tsx (hunk 6 above)
Comment (mrpq) [reply src/iframe-entry/components/Help/Help.tsx]: Done. 2796

File: src/iframe-entry/components/Help/Help.tsx (hunk 5 above)
Comment (mrpq) [reply src/iframe-entry/components/Help/Help.tsx]: checked, and no it's not

File: src/iframe-entry/components/Confirmation.tsx (hunk 4 above)
Comment (mrpq) [reply src/iframe-entry/components/Confirmation.tsx]: <Confirmation> is also being used in contexts where there is no conditions to disable confirmation, so there is no need to pass this prop

**Pull request merged**